primitives/state-machine/src/overlayed_changes.rs

```diff
@@ -498,6 +510,44 @@ impl OverlayedChanges {
 		)
 	}
 
+	/// Get an iterator over all pending and committed child tries in the overlay.
+	pub fn child_infos(&self) -> impl IntoIterator<Item=&ChildInfo> {
+		self.prospective.children_default.iter()
+			.chain(self.committed.children_default.iter())
+			.map(|(_, v)| &v.1).collect::<BTreeSet<_>>()
+	}
+
+	/// Get an iterator over all pending and committed changes.
+	///
+	/// Supplying `None` for `child_info` will only return changes that are in the top
+	/// trie. Specifying some `child_info` will return only the changes in that
+	/// child trie.
+	pub fn changes(&self, child_info: Option<&ChildInfo>)
+		-> impl Iterator<Item=(&StorageKey, &OverlayedValue)>
+	{
+		let (committed, prospective) = if let Some(child_info) = child_info {
+			match child_info.child_type() {
+				ChildType::ParentKeyId => (
+					self.committed.children_default.get(child_info.storage_key()).map(|c| &c.0),
+					self.prospective.children_default.get(child_info.storage_key()).map(|c| &c.0),
+				),
+			}
+		} else {
+			(Some(&self.committed.top), Some(&self.prospective.top))
+		};
+		committed.into_iter().flatten().chain(prospective.into_iter().flatten())
+	}
+
+	/// Return a clone of the currently pending changes.
+	pub fn clone_pending(&self) -> OverlayedChangeSet {
+		self.prospective.clone()
+	}
+
+	/// Replace the currently pending changes.
+	pub fn set_pending(&mut self, pending: OverlayedChangeSet) {
```

Comment:
the function erase pending change, so `replace_pending`, `emplace_pending` or similar name could also be possible.

Reply:
Yes that makes sense. However that pending notion will also go away soon and will just be replaced by a transaction.

Reply:
ok, then it is probably best to use old `prospective` name until future changes.

Reply:
I renamed it to `replace_pending`. I feel like the prospective name is referring strictly to a specific internal layout when other goal is to make the interface more abstract.